src/main/java/net/coderbot/iris/shaderpack/PropertiesPreprocessor.java

```diff
@@ -0,0 +1,140 @@
+
+		for (String line : fileContents.split("\\R")) {
+			String trimmedLine = line.trim();
+
+			if (trimmedLine.startsWith("#if ")) {
+				String[] splitLine = trimmedLine.split(" ");
+
+				if (splitLine.length < 4) continue;
+
+				currentConditional = splitLine[0];
+
+				String variable = splitLine[1];
+				if (MACRO_CONSTANTS.containsKey(variable)) {
+					String operator = splitLine[2];
+					String value = splitLine[3];
+					switch (operator) {
+						case "==":
+							if (value.equals(MACRO_CONSTANTS.get(variable))) {
+								currentlyParsingConditionalProperties = true;
+								continue;
+							}
+							break;
+						case "!=":
+							if (!value.equals(MACRO_CONSTANTS.get(variable))) {
+								currentlyParsingConditionalProperties = true;
+								continue;
+							}
+							break;
+						case ">":
+							try {
+								int intValue = Integer.parseInt(value);
+								int macroIntValue = Integer.parseInt(MACRO_CONSTANTS.get(variable));
+								if (macroIntValue > intValue) {
+									currentlyParsingConditionalProperties = true;
+									continue;
+								}
+							} catch (NumberFormatException e) {
+								Iris.logger.error("Cannot compare non-integer condition value and macro value with > in " + fileName);
+							}
+							break;
+						case ">=":
+							try {
+								int intValue = Integer.parseInt(value);
+								int macroIntValue = Integer.parseInt(MACRO_CONSTANTS.get(variable));
+								if (macroIntValue >= intValue) {
+									currentlyParsingConditionalProperties = true;
+									continue;
+								}
+							} catch (NumberFormatException e) {
+								Iris.logger.error("Cannot compare non-integer condition value and macro value with >= in " + fileName);
+							}
+							break;
+						case "<":
+							try {
+								int intValue = Integer.parseInt(value);
+								int macroIntValue = Integer.parseInt(MACRO_CONSTANTS.get(variable));
+								if (macroIntValue < intValue) {
+									currentlyParsingConditionalProperties = true;
+									continue;
+								}
+							} catch (NumberFormatException e) {
+								Iris.logger.error("Cannot compare non-integer condition value and macro value with < in " + fileName);
+							}
+							break;
+						case "<=":
+							try {
+								int intValue = Integer.parseInt(value);
+								int macroIntValue = Integer.parseInt(MACRO_CONSTANTS.get(variable));
+								if (macroIntValue <= intValue) {
+									currentlyParsingConditionalProperties = true;
+									continue;
+								}
+							} catch (NumberFormatException e) {
+								Iris.logger.error("Cannot compare non-integer condition value and macro value with <= in " + fileName);
+							}
+							break;
+						default: {
+							Iris.logger.error("Invalid operator " + operator + " in " + fileName);
+							continue;
+						}
+					}
+				} else {
+					Iris.logger.warn("Unknown variable fileName " + variable + " in " + fileName);
+					currentlyParsingConditionalProperties = false;
+				}
```

Comment:
Should probably have a separate function for parsing the expression within the `#if` directive and returning whether or not it passed, then that would make it easier to support `#elif`.